app/exerlog/lib/src/core/base/shared_preference/shared_pref_wrapper.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:convert';
+
+import 'package:exerlog/src/core/base/extensions/string_extension.dart';
+import 'package:exerlog/src/utils/logger/logger.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+late SharedPreferences sharedPreferences;
+
+Future<void> initializeSharedPreference() async {
+  sharedPreferences = await SharedPreferences.getInstance();
+}
+
+/// Returns SharedPref Instance
+Future<SharedPreferences> getSharedPref() async {
+  return await SharedPreferences.getInstance();
+}
+
+/// Add a value in SharedPref based on their type - Must be a String, int, bool, double, Map<String, dynamic> or StringList
+Future<bool> setValue(String key, dynamic value) async {
```

Comment:
Maybe instead of having to pass the key here, and having the ability to set any value with this method, we could just make method for every attribute we want to change for example when updating the user we could have something like: 

```
Future<void> setUser(User user) => sharedPreferences.setString(USER_UID, jsonEncode(user.toJson());

Future<User> get user => sharedPreferences.getString(USER_UID);
```
This would take more time to implement because we'd have to do this for every value we want to store. But I think it gives a better overview of all the values that are stored and is also easier to use. what do you think?

Reply:
I didn't think that far. I am thinking more like it's just a wrapper around shared preference and makes things easier. Plus, I think we only need few things stored in the app. For example:

1. If the app is new install (there is a firebase bug, where if you uninstall the app while logged in, it creates some issue when you install the app again)
2. Firebase user uid
3. Also may be a variable to check if user is logged in.

I don't think we need to cache anything. But I might be wrong.

Reply:
Yeah true we don't really have to store a lot, but I prefer this approach because it makes it easier to use and minimises the chance of making mistakes, which is especially important in a open source setting I think. People can for example give the wrong key to this method, or not realise that they're already keys predefined and just give their own string to this method. This should be caught in the PR process but it's a little hard to be sure everything gets checked properly when a lot of people are gonna be working on the project.

Reply:
@jorre127 I resolve this issue for now. But we should open a issue to keep reference of this. Leaving that to you.
@KalleHallden can we use the github discussion tab for this kind of discussion? Easier to keep track